objects/ogamemanager/Step_0.gml

```diff
@@ -1,113 +1,157 @@
 /// @description Insert description here
 // You can write your code in this editor
 
-
-
-// A player is defeated
-if (p1.hp == 0 || p2.hp == 0) 
+switch global.gameMode
 {
-	global.gameHalt = true;
-	gameHaltTimer++;
-	
-	if (gameHaltTimer == 90)
+	case GAMEMODE.VERSUS:
 	{
-		ResetGame();
+		// A player is defeated
+		if (p1.hp == 0 || p2.hp == 0) 
+		{
+			global.gameHalt = true;
+			gameHaltTimer++;
+	
+			if (gameHaltTimer == 90)
+			{
+				ResetGame();
 		
-		SetupGame();
+				SetupGame();
 		
-		global.gameHalt = 0;
-		gameHaltTimer = 0;
-	}
-	if (gameHaltTimer == 1)
-	{
-		if (p1.hp > 0)
-		{
-			global.p1Rounds++;
-		}
-		if (p2.hp > 0)
-		{
-			global.p2Rounds++;
-		} 
-		var particle = instance_create_layer(global.camObj.x-80, 0, "KO_Text", oParticles);
-		with (particle) 
-		{
-		sprite_index = sKOText;
-		image_index = true;
-		lifetime = 89;
+				global.gameHalt = 0;
+				gameHaltTimer = 0;
+			}
+			if (gameHaltTimer == 1)
+			{
+				if (p1.hp > 0)
+				{
+					global.p1Rounds++;
+				}
+				if (p2.hp > 0)
+				{
+					global.p2Rounds++;
+				} 
+				var particle = instance_create_layer(global.camObj.x-80, 0, "KO_Text", oParticles);
+				with (particle) 
+				{
+				sprite_index = sKOText;
+				image_index = true;
+				lifetime = 89;
+				}
+			}
 		}
-	}
-}
 
-// Time runs out
-if (global.gameTimer == 0)
-{
-	global.gameHalt = true;
-	gameHaltTimer++;
+		// Time runs out
+		if (global.gameTimer == 0)
+		{
+			global.gameHalt = true;
+			gameHaltTimer++;
 	
-	if (gameHaltTimer == 90)
-	{
-		ResetGame();
+			if (gameHaltTimer == 90)
+			{
+				ResetGame();
 		
-		SetupGame();
+				SetupGame();
 		
-		global.gameHalt = 0;
-		gameHaltTimer = 0;
-	}
-	if (gameHaltTimer == 1)
-	{
-		if (p1.hp > p2.hp)
-		{
-			global.p1Rounds++;
+				global.gameHalt = 0;
+				gameHaltTimer = 0;
+			}
+			if (gameHaltTimer == 1)
+			{
+				if (p1.hp > p2.hp)
+				{
+					global.p1Rounds++;
+				}
+				if (p2.hp > p1.hp)
+				{
+					global.p2Rounds++;
+				}
+		
+				// Display Time's up text
+				var particle = instance_create_layer(global.camObj.x-80, 0, "KO_Text", oParticles);
+				with (particle) 
+				{
+				sprite_index = sTimeUp;
+				image_index = true;
+				lifetime = 89;
+				}
+			}
 		}
-		if (p2.hp > p1.hp)
+
+		// handle intros
+		if (p1.hasPerformedIntro && p2.hasPerformedIntro && !global.hasCompletedIntros) 
 		{
-			global.p2Rounds++;
+			global.hasCompletedIntros = true;
+			var particle = instance_create_layer(80, 0, "Particles", oParticles);
+			with (particle) 
+			{
+				sprite_index = sRound1Start;
+				lifetime = 110;
+			}
 		}
-		
-		// Display Time's up text
-		var particle = instance_create_layer(global.camObj.x-80, 0, "KO_Text", oParticles);
-		with (particle) 
+
+
+
+		// Frame-by-frame
+		if (keyboard_check_pressed(vk_tab)) or (global.frameskip < 0)

```

Comment:
Use the logical or operator ```||``` instead of the keyword ```or``` and remove the extra parentheses.

Should be: ```if (keyboard_check_pressed(vk_tab) || global.frameskip < 0)```